src/languages/python.ts

```diff
@@ -79,7 +82,10 @@ const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
     ),
     patternMatcher("return_statement.~return!")
   ),
-  argumentOrParameter: argumentMatcher("parameters", "argument_list"),
+  argumentOrParameter: cascadingMatcher(
+    argumentMatcher("parameters", "argument_list"),
+    matcher(patternFinder("generator_expression"), childRangeSelector())
```

Comment:
I get the feeling that this one will match any generator expression. So eg it would match the right-hand side of the following:

```python
foo = (bar for baz in bongo)
```

But that's not actually an argument. So I believe you'll need to look at the parent node to make sure you're in an argument list

Reply:
Since `generator_expression` is a type of argument, coming from call, would it be `call.arguments[generator_expression]`?

https://github.com/tree-sitter/tree-sitter-python/blob/master/src/node-types.json#L776-L785

Reply:
Just confirmed this issue.

Reply:
Would the best path forward be to write my own finder that essentially extends patternFinder() to filter for patterns with the correct parent node type?

Reply:
Or could existing finders such as ancestorChainNodeFinder() perform this functionality if configured correctly?

Reply:
Right after asking I started trying it myself and it seems to work better than I expected. I sort of assumed my first thought wouldn't work.